streaming/streaming.go

```diff
@@ -59,6 +63,23 @@ var (
 	MIN_WORKER_LIMIT int32 = 1
 	// Default gcs path of the Dataflow template.
 	DEFAULT_TEMPLATE_PATH string = "gs://dataflow-templates-southamerica-west1/2023-09-12-00_RC00/flex/Cloud_Datastream_to_Spanner"
+
+	DEFAULT_DATASTREAM_CLIENT_BACKOFF_BASE_DELAY time.Duration = 1.0 * time.Second
+	DEFAULT_DATASTREAM_CLIENT_BACKOFF_MAX_DELAY  time.Duration = 900 * time.Second
+	DEFAULT_DATASTREAM_CLIENT_BACKOFF_MULTIPLIER float64       = 1.6
+	DEFAULT_DATASTREAM_RETRY_CODES               []codes.Code  = []codes.Code{
+		codes.DeadlineExceeded,
+		codes.Unavailable,
+		codes.ResourceExhausted,
+		codes.Unknown,
+	}
```

Comment:
I wonder if we should just let the clients retry for all errors 🤔

HB is not a continuous load generating service where a smaller number of additional retries (even for non-retryable errors) would cause brownouts or outages in the dependent services.

On the other hand, we win in resiliency if we get an undeclared retryable exception from the downstream (what happened in the most recent run with Datastream).

WDYT @VardhanThigle and @bharadwaj-aditya?